bluemira/power_cycle/net.py

```diff
@@ -0,0 +1,870 @@
+        if isinstance(load_type, str):
+            load_type = LoadType[load_type.upper()]
+        return interp1d(
+            self.time[load_type],
+            self.data[load_type],
+            kind=self.model.value,
+            bounds_error=False,  # turn-off error for out-of-bound
+            fill_value=(0, 0),  # below-/above-bounds extrapolations
+        )(time if self.normalised or end_time is None else np.array(time) * end_time)
+
+
+def interpolate_extra(vector: npt.NDArray, n_points: int):
+    """
+    Add points between each point in a vector.
+    """
+    if n_points == 0:
+        return vector
+
+    return np.concatenate([
+        *(
+            np.linspace(vector[s], vector[s + 1], n_points + 1, endpoint=False)
+            for s in range(len(vector) - 1)
+        ),
+        np.atleast_1d(vector[-1]),
+    ])
+
+
+def _normalise_timeseries(
+    time: npt.ArrayLike,
+    end_time: Optional[float] = None,
+) -> Tuple[np.ndarray, Optional[float]]:
+    time = np.asarray(time)
+    if min(time) < 0:
+        raise NotImplementedError("Negative time not supported")
+
+    if max(time) > 1:
+        mx_time = max(time)
+        return time / mx_time, mx_time if end_time is None else end_time
+    return time, end_time
+
+
+class LoadSet:
+    """LoadSet of a phase"""
+
+    def __init__(
+        self,
+        loads: Dict[str, LoadConfig],
+    ):
+        self._loads = loads
+
+    @staticmethod
+    def _consumption_flag(consumption: Optional[bool] = None) -> Set[bool]:
+        return {True, False} if consumption is None else {consumption}
+
+    def get_load_data_with_efficiencies(
+        self,
+        timeseries: np.ndarray,
+        load_type: Union[str, LoadType],
+        unit: Optional[str] = None,
+        end_time: Optional[float] = None,
+        *,
+        consumption: Optional[bool] = None,
+    ) -> Dict[str, np.ndarray]:
+        """
+        Get load data taking into account efficiencies and consumption
+
+        Parameters
+        ----------
+        timeseries:
+            time array
+        load_type:
+            Type of load
+        unit:
+            return unit, defaults to [W] or [var]
+        end_time:
+            for unnormalised loads this assures the load is
+            applied at the right point in time
+        consumption:
+            return only consumption loads
+        """
+        data = self.get_explicit_data_consumption(
+            timeseries, load_type, unit, end_time, consumption=consumption
+        )
+        load_check = LoadType.check(load_type)
+        for ld_name in data:
+            load_conf = self._loads[ld_name]
+            for eff in load_conf.efficiencies:
+                for eff_type, eff_val in eff.value.items():
+                    if eff_type in load_check:
+                        c_eff = (
+                            1 / eff_val
+                            if load_conf.consumption and eff.value > 0
+                            else eff_val
```

Comment:
I come back to the point I mentioned in the general comments and during our discussion: perhaps an implementation independent of the `consumption` input is easier to understand/remember. Production values being lowered and consumption values being raised by efficiencies makes sense from the physical point of view, but the input file will become confusing.

Reply:
Done